docs_src/security/Ch-Secure-Consul.md

```diff
@@ -0,0 +1,82 @@
+# Secure Consul
+
+## Introduction
+
+In the current EdgeX architecture, `Consul` is pre-wired as the default agent service  mainly for
+`Service Configuration`, `Service Registry`, and `Service Health Check` purposes. Prior to EdgeX's
+Ireland release, the communication and configuration to `Consul` are in plain HTTP calls without any
+access control (ACL) token header and thus are insecure.  With the Ireland release,
+that situation is now improved by adding required ACL token header `X-Consul-Token` in any HTTP calls.
+Moreover, `Consul` itself is now bootstrapped and started with its ACL system enabled and thus provides
+better authentication and authorization security features for services.
+
+In this documentation, we will highlight some major features incorporated into EdgeX framework system for
+`Securing Consul`, including how the `Consul` token is generated via the integration of secret store
+management system Vault with Consul system via Vault's Consul Secrets Engine APIs.
+Also briefly give an overview on how Consul token is governed by Vault using Consul's ACL policy associated with
+a Vault role for that token.  An easy way for getting a Consul token from `edgex-compose`'s `compose-builder`
+utility is also given for better developer experiences.
+
+## Consul access token with Vault integration
+
+In order to reduce another token generation system to maintain, we utilize the Vault's feature of
+`Consul Secrets Engine` APIs, governed by Vault itself, and they are integrated with Consul.
+Consul service itself provides ACL system and is enabled via Consul's configuration settings like:
+
+```hcl
+acl = {
+    enabled = true
+    default_policy = "deny"
+    enable_token_persistence = true
+}
+```
+
+and this is set as part of EdgeX `security-bootstrapper` service's process.
+The first main step of `security-bootstrapper` for Consul is to bootstrap Consul's ACL system.
+Once Consul's ACL is successfully bootstrapped, `security-bootstrapper` stores the Consul's ACL bootstrap token
+onto the pre-configured folder under `/tmp/edgex/consul/acl-bootstrap-token` or something like that.
```

Comment:
```suggestion
onto the pre-configured folder under `/tmp/edgex/consul/acl-bootstrap-token`.
```

Reply:
Is that the right path?

Reply:
not sure, need to double check. i put this on my top of head.